dalm/training/generator_only/trainer.py

```diff
@@ -0,0 +1,274 @@
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model_name", type=str, default="HuggingFaceH4/zephyr-7b-alpha", help="the model name")
+    parser.add_argument("--log_with", type=str, default="wandb", help="use 'wandb' to log with wandb")
+    parser.add_argument(
+        "--dataset_name", type=str, default="arcee-ai/azure-reading-comprehension-dataset", help="the dataset name"
+    )
+    parser.add_argument("--split", type=str, default="train", help="the split to use")
+    parser.add_argument("--size_valid_set", type=int, default=4000, help="the size of the validation set")
+    parser.add_argument("--streaming", type=bool, default=False, help="whether to stream the dataset")
+    parser.add_argument("--shuffle_buffer", type=int, default=5000, help="the shuffle buffer size")
+    parser.add_argument("--seq_length", type=int, default=2600, help="the sequence length")
+    parser.add_argument("--num_workers", type=int, default=4, help="the number of workers")
+
+    parser.add_argument("--eval_steps", type=int, default=200, help="the evaluation frequency")
+    parser.add_argument("--num_train_epochs", type=int, default=3, help="the number of training epochs")
+    parser.add_argument("--logging_steps", type=int, default=10, help="the logging frequency")
+    parser.add_argument("--per_device_train_batch_size", type=int, default=1, help="the per device train batch size")
+    parser.add_argument("--per_device_eval_batch_size", type=int, default=1, help="the per device eval batch size")
+    parser.add_argument("--gradient_accumulation_steps", type=int, default=32, help="the gradient accumulation steps")
+    parser.add_argument(
+        "--gradient_checkpointing", type=bool, default=True, help="whether to use gradient checkpointing"
+    )
+    parser.add_argument("--group_by_length", type=bool, default=False, help="whether to group by length")
+    parser.add_argument("--packing", type=bool, default=True, help="whether to use packing for SFTTrainer")
+
+    parser.add_argument("--lora_alpha", type=float, default=512, help="the lora alpha parameter")
+    parser.add_argument("--lora_dropout", type=float, default=0.05, help="the lora dropout parameter")
+    parser.add_argument("--lora_r", type=int, default=256, help="the lora r parameter")
+
+    parser.add_argument("--learning_rate", type=float, default=1e-4, help="the learning rate")
+    parser.add_argument("--lr_scheduler_type", type=str, default="cosine", help="the lr scheduler type")
+    parser.add_argument("--num_warmup_steps", type=int, default=100, help="the number of warmup steps")
+    parser.add_argument("--weight_decay", type=float, default=0.05, help="the weight decay")
+    parser.add_argument("--optimizer_type", type=str, default="paged_adamw_32bit", help="the optimizer type")
+
+    parser.add_argument("--output_dir", type=str, default="./generator_finetuned_model", help="the output directory")
```

Comment:
I think we should add this directory to `.gitignore` so that it doesn't show up in the list of changed files.

Reply:
It wouldn't show up in the list of changes files. It would show up in the list of untracked files no?

Reply:
Yeah but that's just as bad.  Users that want to contribute would be confused by the untracked files being created when they ran `git status`.  Also it would be a nuisance to core committers to have ignore them when running `git add`